server/src/main/scala/org/http4s/server/middleware/RequestLogger.scala

```diff
@@ -0,0 +1,29 @@
+package org.http4s
+package server
+package middleware
+
+import fs2._
+import org.log4s._
+import cats.implicits._
+import fs2.interop.cats._
+
+/**
+  * Simple Middleware for Logging Requests As They Are Processed
+  */
+object RequestLogger {
```

Comment:
We could have a logger, that we enable different types of Logger internally. However this could get out of hand very quickly, however I agree this is not an extensible enough solution.